exchanges/kucoin/kucoin.go

```diff
@@ -0,0 +1,2047 @@
+package kucoin
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/common/crypto"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/kline"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/order"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/orderbook"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/request"
+)
+
+// TODO:
+// run linter
+// handle rate limit for all API
+
+// Kucoin is the overarching type across this package
+type Kucoin struct {
+	exchange.Base
+}
+
+const (
+	kucoinAPIURL        = "https://api.kucoin.com/api"
+	kucoinWebsocketURL  = "wss://ws-api.kucoin.com/endpoint" // Prone to change
```

Comment:
Whenever you want to connect to a WebSocket server you send an HTTP Post request to retrieve the WebSocket server endpoint. I will remove this line since it won't be used.